spot_rl_experiments/spot_rl/utils/geometry_utils.py

```diff
@@ -134,3 +137,117 @@ def is_pose_within_bounds(
     )
 
     return bool(is_within_linear_bounds and is_within_angular_bounds)
+
+
+def is_position_within_bounds(
+    test_position, target_position, xy_dist_threshold, z_dist_threshold
+) -> bool:
+    """
+    Check if a test position(x,y,z) is within  of target pose.
+
+    Args:
+        test_position (list): Test position, as [x,y,z]
+        target_position (list): Target position, as [x,y,z]
+        xy_dist_threshold (float): Threshold for xy distance (in meters)
+        z_dist_threshold (float): Threshold for z distance (in meters)
+
+    Returns:
+        is_within (bool): True if the test pose is within both linear and angular bounds of the target pose, False otherwise.
+    """
+    # Linear xy dist bounds
+    is_within_linear_xy_bounds = (
+        euclidean(test_position[:2], target_position[:2]) < xy_dist_threshold
+    )
+
+    # Linear z dist bounds
+    is_within_linear_z_bounds = (
+        abs(test_position[2] - target_position[2]) < z_dist_threshold
+    )
+
+    return bool(is_within_linear_xy_bounds and is_within_linear_z_bounds)
+
+
+def interpolation_between_two_positions(point1, point2, steps, kind="linear"):
+    """
+    Interpolate between two 3D points
+
+    Args:
+        point1 (tuple): The first 3D point
+        point2 (tuple): The second 3D point
+        steps (int): The number of steps to interpolate
+        kind (str): The type of interpolation to use (default: linear)
+
+    Returns:
+        interpolated_points (List[Tuple]): A list of interpolated points
+    """
+    # Define the two 3D points
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+
+    # Create an array of x,y,z values for interpolation
+    x = np.array([x1, x2])
+    y = np.array([y1, y2])
+    z = np.array([z1, z2])
+
+    # Create a 1d interpolation object object for each dimension
+    f_x = interp1d(x, y, kind=kind)
+    f_y = interp1d(x, y, kind=kind)
+    f_z = interp1d(x, z, kind=kind)
+
+    # Generate the interpolated points
+    interpolated_points = []  # type: List[Tuple]
+    for t in np.linspace(x1, x2, steps):
+        interpolated_x = float(f_x(t))
+        interpolated_y = float(f_y(t))
+        interpolated_z = float(f_z(t))
+        interpolated_points.append((interpolated_x, interpolated_y, interpolated_z))
+
+    # Return the interpolated points
+    return interpolated_points
+
+
+def generate_intermediate_point(point1, point2):
+    """
+    Generate an intermediate point between two 3D points
+
+    Args:
+        point1 (tuple): The first 3D point
+        point2 (tuple): The second 3D point
+
+    Returns:
+        intermediate_point (tuple): The intermediate 3D point
+    """
+    # Define the two 3D points
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+
+    # Elevate the z coordinate
+    z_elevation = 0.2
```

Comment:
make this input instead of magic number set in the function

Reply:
added